pkg/odo/cli/component/common_link.go

```diff
@@ -357,6 +354,8 @@ func (o *commonLinkOptions) validateForOperator() (err error) {
 			return fmt.Errorf("couldn't find service named %q. Refer %q to see list of running services", svcFullName, "odo service list")
 		}
 	} else {
+		o.serviceType = "Service"
+		svcFullName = o.serviceName
 		if o.suppliedName == o.EnvSpecificInfo.GetName() {
 			return fmt.Errorf("the component %q cannot be linked with itself", o.suppliedName)
```

Comment:
This is probably not related to this PR, but it seems to me like a minor fix.
```
➜  comp odo unlink nodejs-comp-oelo                             
 ✗  the component "nodejs-comp-oelo" cannot be linked with itself
```
I was playing with odo and ended up unlinking a component from itself, which threw the above message, `cannot be linked`, even though I was trying to unlink. Since `commonLinkOptions` is used by both link and unlink, maybe we can fix this message to be more generic.

Reply:
Thanks, done